lib/workflow/stores/mongo.js

```diff
@@ -985,5 +985,82 @@ function mongoStoreFactory(waterline, Promise, Constants, Errors, assert, _, san
         return waterline.graphobjects.findOneMongo(query);
     };
 
+    /**
+     * Update graph progress
+     * @param {Object} data - data includes graphId, progress message
+     * @returns {Promise} a promise with modified percentage info of next graph progress data 
+     * @memberOf store
+     */
+    exports.updateGraphProgress = function(data) {
+        assert.uuid(data.graphId, "Graph ID should be a UUID");
+        assert.string(data.progress.description, "Progress info should include description");
+        assert.string(data.progress.percentage, "Progress info should include percentage data");
+        var update;
+        var query = {
+            instanceId: data.graphId,
+            _status: {$in: Constants.Task.ActiveStates}
+        };
+        var options = {
+            multi: true
+        };
+        return waterline.graphobjects.findOne({instanceId: data.graphId})
+        .then(function(graphObject){
+            if (!_.has(data, 'graphName')) {
+                data.graphName = graphObject.definition.friendlyName;
```

Comment:
Just to align with taskProgress. Both task and workflow progress will includes task/workflow name in AMQP message.